src/Promise.php

```diff
@@ -65,7 +65,7 @@ public function getState();
      *
      * @param bool $unwrap Whether to return resolved value / throw reason or not
      *
-     * @return T Resolved value, null if $unwrap is set to false
+     * @return ($unwrap is true ? T : null|void) Resolved value, null if $unwrap is set to false
```

Comment:
how can it return null or void? should this be:
```suggestion
     * @return ($unwrap is true ? T|null : void) Resolved value, or nothing if $unwrap is set to false
```

Reply:
I was guided by https://github.com/php-http/client-common/blob/2.x/src/Deferred.php#L142
Would we need to update client-common to return void instead?

Reply:
afaik its not possible to `return void`. if a method never returns a value, we can use the type declaration `: void` on the method, which makes php realize you can not use the return value. but with this dual use method we can't do that.

if we just `return;` its the same as `return null;` https://3v4l.org/3e3RK

looking at deferred and thinking about this, i think the technically correct doc is

```suggestion
     * @return ($unwrap is true ? T|null : null) Resolved value, null if $unwrap is set to false
```

Reply:
Sorry I meant just `return` instead of `return null`. Interesting didn't realise `return;` implied `return null`, thanks!

`@return ($unwrap is true ? T: null)` should cover a nullable resolve type as well - https://phpstan.org/r/566b7379-3041-438f-86c1-078ee3c95fb8

Reply:
ah, with `@implements Promise<string|null>` the `@return ($unwrap is true ? T : null)` is enough indeed. and thats the correct way to do it - the promise could also be defined to always have a value or throw an exception, and then there can be no null return when unwrap is true.